src/lib.ml

```diff
@@ -22,23 +22,35 @@ end
 include T
 module Set = Set.Make(T)
 
-let dir = function
-  | Internal (dir, _) -> dir
+let dir ~context ~scope = function
+  | Internal (dir, lib) ->
+    if Jbuild.Scope.compare scope lib.scope = 0 then
+      dir
+    else begin
+      match lib.public, lib.scope.name with
+      | Some {sub_dir;_}, Some scope_name ->
+        let install_dir = Config.local_install_dir ~context in
+        Path.relative
+          (Path.append install_dir (Install.lib_install_path ~package:scope_name))
```

Comment:
The name is chosen as the shortest package name so it's deterministic, but each package will install its files in `<libdir>/<package>`